### DISCUSSION TITLE
ALOS PALSAR Level 1.1 (FBD H-H pair) processing using ISCE: Questions regarding misregistration step

### DISCUSSION THREAD

**Discussion by Hassan2211345** [isce-framework/isce2#898] [Q&A]
Hey there everyone,
I have run stripmapApp processing using ISCE, to generate interferogram using a pair of scenes.
Step misregistration gave me some warnings.

Even though it completed to the end successfully, there were warning that I think might be the cause of distortion of my interferogram in the end. My question is, the parameters for example the window skip size, can we include something in our input file to avoid such problem.

Attached is my final figure after running up to the filtered interferogram stage.

Misregistration step processing dialogues:
Scale Factor in Range:  1.0
Scale Factor in Azimuth:  1.0
Skip Sample Down: 924
Skip Sample Across: 99
 Input Bands:            1           1
  XXX start timer
 wisdomFile, length =            0
 WARNING - number of samples availabe for estating gross offset is NOT power of 2:          80
           the value is computed as 2*i_srchy/winsize_filt
           better to make it power of 2 for FFT
           current i_srchy (azimuth search window size):          40
           current winsize_filt (azimuth filtering window size):           1
CAUTION - Requesting very large search window pull in
Reference Window Size is                     128 sample pixels
Number of Search Pixels is                    40 sample pixels
The rule of thumb is that the search window pull in is at least 5
pixels and is less than the reference window size divided by 5. 
Suggested Number of Search Pixels is          21 sample pixels
 
 
CAUTION - Requesting very large search window pull in
Reference Window Size is                     128 line pixels
Number of Search Pixels is                    40 line pixels
The rule of thumb is that the search window pull in is at least 5
pixels and is less than the reference window size divided by 5. 
Suggested Number of Search Pixels is          21 line pixels
 
 
Requested resolving shifts to 1/   32 of a pixel
 
INFORMATION - you choose skips which are small for your window sizes
Normally the skip size is bigger than the box size
Across your skip is         99 but your window is        128
Down   your skip is        924 but your window is        128
This means that the image chips are larger than the separation between chipsAt line =         5853
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 XXX time for inner loop    15.6950455    
 At line =         6777
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 XXX time for inner loop    15.7596283    
 At line =         7701
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 XXX time for inner loop    15.9808502  

.
.
.
 At line =        36345
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 Bad match at level 1
 XXX time for inner loop    15.0672607    
 Elapsed time.    634.106262    
GDAL close: P349I250_slc/P349I250.slc.vrt
GDAL close: coregisteredSlc/coarse_coreg.slc.vrt
*************************************
rgratio, azratio:  1.0 1.0
*************************************
DEBUG %%%%%%%%
1109
DEBUG %%%%%%%%
2024-11-08 16:05:10,325 - isceobj.Location.OffsetField - INFO - 339 offsets culled

 1-D calculation:

          Slope across  Intercept:
 Across:   -8.82109103E-04   2.50809312
 Down:      2.27697135E-04  -4.27282476

60 points left
DEBUG %%%%%%%%
60
DEBUG %%%%%%%%
2024-11-08 16:05:10,329 - isceobj.Location.OffsetField - INFO - 0 offsets culled

 1-D calculation:

          Slope across  Intercept:
 Across:    1.29267210E-04   1.13520265
 Down:      2.32465580E-04  -5.98970604

17 points left
DEBUG %%%%%%%%
17
DEBUG %%%%%%%%
2024-11-08 16:05:10,331 - isceobj.Location.OffsetField - INFO - 0 offsets culled

 1-D calculation:

          Slope across  Intercept:
 Across:    4.20962286E-04   1.46670210
 Down:      3.12891352E-04  -7.42067003

7 points left
DEBUG %%%%%%%%
7
DEBUG %%%%%%%%
2024-11-08 16:05:10,333 - isceobj.Location.OffsetField - INFO - 0 offsets culled

 1-D calculation:

          Slope across  Intercept:
 Across:    1.15607120E-03 -0.492547482
 Down:      5.57098072E-04  -11.3835640

1 points left
No chi squared value....
Try reducing rank of polynomial.
No chi squared value....
Try reducing rank of polynomial.
No chi squared value....
Try reducing rank of polynomial.
No chi squared value....
Try reducing rank of polynomial.
Estimated az shift:  1.03125
Estimated rg shift:  3.0625

```typescript
<stripmapApp>
<component name="insar">
<property name="sensor name">ALOS</property>
<property name="range looks">4</property>
<property name="azimuth looks">16</property>
<component name="reference">
<catalog>reference.xml</catalog>
</component>
<component name="secondary">
<catalog>secondary.xml</catalog>
</component>
<property name="filter strength">0.7</property>
<property name="do unwrap">False</property>
<property name="unwrapper name">snaphu</property>
<property name="do split spectrum">False</property>
<property name="do dispersive">False</property>
</component>
</stripmapApp>
``` 

Please note, the number of looks in range and azimuth were included just prior to the interferogram step.
I will appreciate your feedback on this and thanks in advance..

![interferogram_plots](https://github.com/user-attachments/assets/90a16ce0-8d07-49a8-a63f-d35ebc8a6db2)

**Comment by keepfighting666** [comment]
Hello, can you share you workflow, I have no idea to process ALOS 1 level 1.1 data. Thanks a lot.

**Reply by Hassan2211345** [reply]
Hi, I think with ISCE, its a good idea to start with level 1.0 data. I am using level 1.0 data too

**Comment by keepfighting666** [comment]
I've tried, but there is only 2 or 3 level 1.0 images in my ROI.

Hassan2211345 ***@***.***> 于2024年12月11日周三 07:27写道：
